mantidproject/mantid
# All recent changes to Indirect Inelastic including dependency change

### Description of work
A large number of PRs being targeted at `main` are periodically failing due to the following warning
```
08:04:25 ninja: error: dependency cycle: bin/libMantidScientificInterfacesInelasticQt5.so -> qt/scientific_interfaces/Inelastic/CMakeFiles/MantidScientificInterfacesInelasticQt5.dir/Manipulation/InelasticDataManipulation.cpp.o -> /jenkins_workdir/workspace/pull_requests-conda-linux/build/qt/scientific_interfaces/Indirect/qt5/ui_IndirectInstrumentConfig.h -> bin/libMantidScientificInterfacesInelasticQt5.so
```

I believe this is caused by a change in which the order that the Inelastic and Indirect interface libraries depend on each other was swapped in https://github.com/mantidproject/mantid/pull/36603. It used to be that Inelastic depended on Indirect, but now its the other way around. I think when we do not do a `clean` build, it is using some previous build files from before this change, and hence thinks there is a circular dependency. 

Why does this keep cropping up? My current theory is that `ornl-next` does not have the changes, so whenever they do a build it creates outdated build files, which then cause a conflict when the next time a PR runs against main.

Indirect and Inelastic has changed quite a lot this release that it is near impossible to reconcile `main` and `ornl-next` without just copying and pasting the new Indirect and Inelastic folders over. Hence the diff is so large because a lot of files in the Indirect and Inelastic folders, if not most, have been moved around. 


### To test:
Make sure it builds and passes the tests. Make sure the changes are only in the `Inelastic` and `Indirect` folders, plus allow the changes in `ContourPreviewPlot` and `FunctionModel`.

You can also try check this out locally and build it.

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
